frontend/src/hooks/useCompareRoutes.tsx

```diff
@@ -0,0 +1,80 @@
+import {useEffect, useState} from "react";
+import {Route} from "../models/RouteModel";
+import axios from "axios";
+import {CompareRoutes} from "../models/CompareRoutesModel";
+import {toast} from "react-toastify";
+
+export default function useCompareRoutes() {
+    const [routesToCompare, setRoutesToCompare] = useState<Route[]>([]);
+
+    const initialStateComparedRoutes: CompareRoutes = {
+        id: "",
+        compared: [
+            {
+                id: "",
+                start: "",
+                destination: "",
+                distance: 0,
+                numberOfPersons: 1,
+                oneWay: true,
+                vehicle: {
+                    type: "",
+                    co2Emission: 0,
+                    fuel: "",
+                    carSize: "",
+                    distanceLevel: "",
+                    meansOfTransport: ""
+                },
+                "co2EmissionRoute": 0
+            },
+            {
+                id: "",
+                start: "",
+                destination: "",
+                distance: 0,
+                numberOfPersons: 1,
+                oneWay: true,
+                vehicle: {
+                    type: "",
+                    co2Emission: 0,
+                    fuel: "",
+                    carSize: "",
+                    distanceLevel: "",
+                    meansOfTransport: ""
+                },
+                co2EmissionRoute: 0
+            }
+        ],
```

Comment:
Twice the same code could be simplified with something like that (not sure about the Route type):
```suggestion
     initRoute: Route = {
                 id: "",
                 start: "",
                 destination: "",
                 distance: 0,
                 numberOfPersons: 1,
                 oneWay: true,
                 vehicle: {
                     type: "",
                     co2Emission: 0,
                     fuel: "",
                     carSize: "",
                     distanceLevel: "",
                     meansOfTransport: ""
                 },
                 co2EmissionRoute: 0
             }
             
        compared: [initRoute, initRoute ],
```